src/main/java/org/candlepin/subscriptions/security/OptInChecker.java

```diff
@@ -46,6 +46,11 @@ public OptInChecker(OrgConfigRepository orgConfigRepository) {
 
   public boolean checkAccess(Authentication authentication) {
     Object principal = authentication.getPrincipal();
+    if (RhAssociatePrincipal.class.isAssignableFrom(principal.getClass())
+        || PskClientPrincipal.class.isAssignableFrom(principal.getClass())
```

Comment:
Don't see why the condition `PskClientPrincipal.class.isAssignableFrom(principal.getClass())` was added. I've tried the calls in the PR description, but none exercises this condition.